packages/frontend/src/components/UserSettings/UsersAndGroupsPanel/UsersView.tsx

```diff
@@ -62,28 +62,27 @@ const UserNameDisplay: FC<{
 }> = ({ user, showInviteLink, hasEmail, onGetLink }) => {
     return (
         <Flex justify="space-between" align="center">
-            {user.isActive ? (
-                <Stack spacing="xxs">
-                    <Title order={6}>
-                        {user.firstName} {user.lastName}
+            {!user.isActive ? (
```

Comment:
Flip logic to check first if user is inactive before checking pending status.
+ show email if they don't have names.